### PR TITLE
Added a contributing guide document [9369]

### PR DESCRIPTION
**Pull request by imontesino**
Signed-off-by: imontesino <monte.igna@gmail.com>

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
+# Contribution Guidelines
+
+*Fast DDS* is an open source C++ implementation of the DDS (Data Distribution Service) standard of the OMG (Object
```
Comment (MiguelCompany) [inline CONTRIBUTING.md]: This is for all eProsima open source projects

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+## Submitting a pull request
+
+If writing an issue does not do it for you because you just know what has to be done and how, and you have the energy
+and time to directly contribute to the project, submitting a pull request is your way to go. Please mind that by doing
+so, you accept to abide to the pull request process followed on this repository, meaning:
+
+1. Modify one feature at a time. Avoid pull requests where several features or functionalities are implemented or
+   modified at the same time. If you have many contributions, please split them in different pull requests.
+1. Submit your pull requests in separate branches. As for branch names, we normally use `features/*` for adding
```
Comment (MiguelCompany) [inline CONTRIBUTING.md]:
```suggestion
1. Submit your pull requests in separate branches. As for branch names, we normally use `feature/*` for adding
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+1. Contributors must sign-off each commit by adding a `Signed-off-by: ...` line to commit messages to certify that they
+   have the right to submit the code they are contributing to the project according to the [Developer Certificate of
+   Origin (DCO)](https://developercertificate.org/). Pull request with unsigned commits will not pass the automated
+   checks until unsigned commits have been amended.
+1. Test your changes before submission, and only submit "green" pull requests. Our Continuous Integration will still run
+   the tests, but this way the process becomes smoother and the "Fixing errors" commits and re-runs are avoided. If you
+   need any help setting up, please let us know by opening an issue. We will be happy to help you and to improve each
+   repository's instructions.
+1. Be prepared to receive criticism, and potentially tons of change requests. Please do not take criticism as personal,
+   we really appreciate your contributing effort. To be merged into the master branch o the releases, a pull request
```
Comment (MiguelCompany) [inline CONTRIBUTING.md]:
```suggestion
   we really appreciate your contributing effort. To be merged into the master branch or the releases, a pull request
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+## Coverage and Performance Policies
+
+Coverage tests and performance tests are run in CI on the nightly builds of the master branch. New changes should never
+lower the coverage result of the software. 
+
+If at any time the master branch exhibits a performance regression the priority is to restore to the last working
+version, until the root cause has been identified an proceeded to be fixed. 
+
+
```
Comment (MiguelCompany) [inline CONTRIBUTING.md]: I think this does not belong here

**Review by MiguelCompany — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+   checks until unsigned commits have been amended.
+1. Test your changes before submission, and only submit "green" pull requests. Our Continuous Integration will still run
+   the tests, but this way the process becomes smoother and the "Fixing errors" commits and re-runs are avoided. If you
+   need any help setting up, please let us know by opening an issue. We will be happy to help you and to improve each
+   repository's instructions.
+1. Be prepared to receive criticism, and potentially tons of change requests. Please do not take criticism as personal,
+   we really appreciate your contributing effort. To be merged into the master branch o the releases, a pull request
+   must be approved by at least one reviewer with write access.
+1. Changes of the kind `feature/*` must be accompanied by a pull request to the [Fast-DDS
+   Docs](https://github.com/eProsima/Fast-RTPS-docs) documenting the added functionality.
```
Comment (JLBuenoLopez) [inline CONTRIBUTING.md]: It should not refer to a particular repository documentation. It should be generic.

**Review by JLBuenoLopez — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+If writing an issue does not do it for you because you just know what has to be done and how, and you have the energy
+and time to directly contribute to the project, submitting a pull request is your way to go.
+Please mind that by doing so, you accept to abide to the pull request process followed on this repository, meaning:
+
+1. Modify one feature at a time.
+   Avoid pull requests where several features or functionalities are implemented or modified at the same time.
+   If you have many contributions, please split them in different pull requests.
+1. Submit your pull requests in separate branches. 
+   As for branch names, we normally use `feature/*` for addingfunctionality or refactors, `hotfix/*` for small fixes,
```
Comment (JLBuenoLopez) [inline CONTRIBUTING.md]:
```suggestion
   As for branch names, we normally use `feature/*` for adding functionality or refactors, `hotfix/*` for small fixes,
```

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+   and `bugfix/*` for larger parts that do not reflect reality and need to be corrected.
+1. Code should be formatted according to our [Code Style Guide](https://github.com/eProsima/cpp-style), we also provide
+an [uncrustify](https://github.com/uncrustify/uncrustify) configuration file to help in that regard. 
+1. Contributors must sign-off each commit by adding a `Signed-off-by: ...` line to commit messages to certify that they
+   have the right to submit the code they are contributing to the project according to the [Developer Certificate of
+   Origin (DCO)](https://developercertificate.org/).
+   Pull request with unsigned commits will not pass the automated checks until unsigned commits have been amended.
```
Comment (JLBuenoLopez) [inline CONTRIBUTING.md]:
```suggestion
   Pull requests with unsigned commits will not pass the automated checks until unsigned commits have been amended.
```

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+   Avoid pull requests where several features or functionalities are implemented or modified at the same time.
+   If you have many contributions, please split them in different pull requests.
+1. Submit your pull requests in separate branches. 
+   As for branch names, we normally use `feature/*` for addingfunctionality or refactors, `hotfix/*` for small fixes,
+   and `bugfix/*` for larger parts that do not reflect reality and need to be corrected.
+1. Code should be formatted according to our [Code Style Guide](https://github.com/eProsima/cpp-style), we also provide
+an [uncrustify](https://github.com/uncrustify/uncrustify) configuration file to help in that regard. 
+1. Contributors must sign-off each commit by adding a `Signed-off-by: ...` line to commit messages to certify that they
+   have the right to submit the code they are contributing to the project according to the [Developer Certificate of
+   Origin (DCO)](https://developercertificate.org/).
```
Comment (JLBuenoLopez) [inline CONTRIBUTING.md]: Too long. Maybe consider splitting this sentence in two.

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+
+Most of eProsima's products follow an open source model.
+As such, we are always listening to the community for issues, feature improvements and suggestions.
+
+
+## Submitting a pull request
+
+If writing an issue does not do it for you because you just know what has to be done and how, and you have the energy
+and time to directly contribute to the project, submitting a pull request is your way to go.
+Please mind that by doing so, you accept to abide to the pull request process followed on this repository, meaning:
```
Comment (JLBuenoLopez) [inline CONTRIBUTING.md]:
```suggestion
Please mind that by doing so, you accept to abide to the pull request process followed by eProsima, meaning:
```
These rules do not apply only to this repository, but also to all eProsima repositories

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+
+1. Modify one feature at a time.
+   Avoid pull requests where several features or functionalities are implemented or modified at the same time.
+   If you have many contributions, please split them in different pull requests.
+1. Submit your pull requests in separate branches. 
+   As for branch names, we normally use `feature/*` for addingfunctionality or refactors, `hotfix/*` for small fixes,
+   and `bugfix/*` for larger parts that do not reflect reality and need to be corrected.
+1. Code should be formatted according to our [Code Style Guide](https://github.com/eProsima/cpp-style), we also provide
```
Comment (JLBuenoLopez) [inline CONTRIBUTING.md]:
```suggestion
1. Code should be formatted according to our [Code Style Guide](https://github.com/eProsima/cpp-style). We also provide
```

File: CONTRIBUTING.md [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+an [uncrustify](https://github.com/uncrustify/uncrustify) configuration file to help in that regard. 
+1. Contributors must sign-off each commit by adding a `Signed-off-by: ...` line to commit messages to certify that they
+   have the right to submit the code they are contributing to the project according to the [Developer Certificate of
+   Origin (DCO)](https://developercertificate.org/).
+   Pull request with unsigned commits will not pass the automated checks until unsigned commits have been amended.
+1. Test your changes before submission, and only submit "green" pull requests.
+   Our Continuous Integration will still run the tests, but this way the process becomes smoother and the "Fixing
+   errors" commits and re-runs are avoided.
+   If you need any help setting up, please let us know by opening an issue.
+   We will be happy to help you and to improve each repository's instructions.
```
Comment (JLBuenoLopez) [inline CONTRIBUTING.md]:
```suggestion
   We will be happy to help you and to improve the instructions for each repository if needed.
```

**Review by JLBuenoLopez — COMMENTED** [review]
Please, consider these comments

**Review by JLBuenoLopez — APPROVED** [review]
LGTM

**Review by MiguelCompany — APPROVED** [review]

**Pull request merged**